pkg/sql/catalog/tabledesc/table_desc.go

```diff
@@ -183,6 +184,33 @@ func UpdateIndexPartitioning(
 	idx.KeyColumnNames = append(newColumnNames, idx.KeyColumnNames[oldNumImplicitCols:]...)
 	idx.KeyColumnDirections = append(newColumnDirections, idx.KeyColumnDirections[oldNumImplicitCols:]...)
 	idx.Partitioning = newPartitioning
+	if !isIndexPrimary {
+		return true
+	}
+
+	newStoreColumnIDs := make([]descpb.ColumnID, 0, len(idx.StoreColumnIDs))
+	newStoreColumnNames := make([]string, 0, len(idx.StoreColumnNames))
+	for i := range idx.StoreColumnIDs {
+		id := idx.StoreColumnIDs[i]
+		name := idx.StoreColumnNames[i]
+		found := false
+		for _, newColumnName := range newColumnNames {
```

Comment:
why do this search by name and not by ID? I don't recall how `ALTER COLUMN TYPE` works exactly but I recall a new column being added with the same name and then being swapped it. I don't remember if we use a temp name.

Reply:
I can't remember but it might have been the case that I thought (mistakenly or not) that some IDs might still be zero at this point. I hope I'm wrong. This just reinforces my feeling that this stuff is undertested.

Reply:
Confirming that this needs to be by name because this can be called pre-ID-allocation.